packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,38 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for a post-blob activated chain.
+    /// Params are passed in as part of msg.data in order to compress the calldata.
+    /// Params should be passed in in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _number             L1 blocknumber.
+    ///   5. _timestamp          L1 timestamp.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+	    //require(msg.sender == DEPOSITOR_ACCOUNT, "L1Block: only the depositor account can set L1 block values");
+	    assembly {
+            // Revert if the caller is not the depositor account.
+            if xor(caller(), DEPOSITOR_ACCOUNT) {
+                mstore(0x00, 0x3cc50b45)
+                revert(0x1C, 0x04)
+            }
+            let data := calldataload(4)
+            sstore(basefeeScalar.slot, shr(192, data)) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
+            // sequenceNumber is a 64 bit int that has a 256-bit slot to itself, so we must clear
+            // out the scalar bits to ensure the topmost 192 bits in the slot will always be 0.
+            sstore(sequenceNumber.slot, and(shr(128, data), 0xFFFFFFFFFFFFFFFF)) // uint64
+            sstore(sequenceNumber.slot, shr(192, shl(64, data))) // uint64
```

Comment:
Is this right? I might not be following properly, but it looks like we write to `sequenceNumber.slot` and then immediately write over it again.

Reply:
sorry that was a bad code merge, that second line was an older version of the one above it.  Removed.

Reply:
Got it, FWIW I don't think they did the same thing either. Maybe the new line was a fix though? Feel free to resolve.

Reply:
it shouldn't have been different, at least fuzz testing seems to think they are doing the same thing?